From 7dce657fb2bcab024c3c26fc242d4cb908c28a8c Mon Sep 17 00:00:00 2001
From: svenate <salini.venate@intel.com>
Date: Thu, 3 Mar 2022 11:46:32 +0530
Subject: [PATCH] WA-Move -pie option to linker flags

Move the -pie option from CFLAGS to LDFLAGS
to fix below build error:
clang++: error: argument unused during compilation:
'-pie' [-Werror,-Wunused-command-line-argument]

Signed-off-by: svenate <salini.venate@intel.com>
Signed-off-by: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
---
 android/mfx_defs.mk       | 4 ++--
 builder/FindGlobals.cmake | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/android/mfx_defs.mk b/android/mfx_defs.mk
index 56e67afb..e4358b28 100644
--- a/android/mfx_defs.mk
+++ b/android/mfx_defs.mk
@@ -77,7 +77,7 @@ MFX_CFLAGS += \
 #  Security
 MFX_CFLAGS += \
   -fstack-protector \
-  -fPIE -fPIC -pie \
+  -fPIE -fPIC \
   -O2 -D_FORTIFY_SOURCE=2 \
   -Wformat -Wformat-security \
   -fexceptions -frtti
@@ -117,7 +117,7 @@ MFX_INCLUDES_LIBVA := $(TARGET_OUT_HEADERS)/libva
 
 # Setting usual link flags
 MFX_LDFLAGS := \
-  -z noexecstack \
+  -z noexecstack -pie \
   -z relro -z now
 
 # Setting vendor
diff --git a/builder/FindGlobals.cmake b/builder/FindGlobals.cmake
index 6c96fc8e..d0d58069 100644
--- a/builder/FindGlobals.cmake
+++ b/builder/FindGlobals.cmake
@@ -71,7 +71,7 @@ if( Linux OR Darwin )
 
   set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} -pipe -fPIC ${c_warnings} ${no_warnings} ${common_flags}")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pipe -fPIC ${cxx_warnings} ${no_warnings} ${common_flags}")
-  append("-fPIE -pie" CMAKE_EXE_LINKER_FLAGS)
+  append("-fPIE" CMAKE_EXE_LINKER_FLAGS)
 
   # CACHE + FORCE should be used only here to make sure that this parameters applied globally
   # End user is responsible to adjust configuration parameters further if needed. Here
-- 
2.17.1

